r/R/dplyr-funcs-conditional.R

```diff
@@ -90,7 +90,15 @@ register_bindings_conditional <- function() {
     out
   })
 
-  register_binding("dplyr::case_when", function(...) {
+  register_binding("dplyr::case_when", function(..., .default = NULL, .ptype = NULL, .size = NULL) {
```

Comment:
> raises errors if non-NULL values are supplied for .ptype and .size.

Would it be better to mention the limitation with the `notes`?

https://github.com/apache/arrow/blob/7f3e87d2d5e6f0568dbb92b468d911717d9aed91/r/R/dplyr-funcs.R#L63-L64